lib/iris/cube.py

```diff
@@ -3021,8 +3021,7 @@ def __eq__(self, other):
             # having checked everything else, check approximate data
             # equality - loading the data if has not already been loaded.
             if result:
-                result = np.all(np.abs(self.data - other.data) < 1e-8)
-
+                result = da.array.allclose(self.data, other.data)
```

Comment:
`da` is already `dask.array` and the function returns a dask array.  So I think you need:
```python
da.allclose(self.data, other.data).compute()
```